public/js/utils/source-editor.js

```diff
@@ -39,6 +39,21 @@ class SourceEditor {
   }
 
   /**
+   * Replaces the current document with a new source document
+   */
+  replaceDocument(doc) {
+    this.editor.swapDoc(doc);
+  }
+
+  /**
+   * Creates a CodeMirror Document
+   * @returns CodeMirror.Doc
+   */
+  createDocument() {
+    return new CodeMirror.Doc("");
+  }
```

Comment:
while not a big concern now, i did keep in mind the editor integration a bit here and believe we can implement similar methods if we integrate with atom, emacs, etc